projects/fusion-e2e/tests/components/alert/alert-component.ts

```diff
@@ -1,45 +1,59 @@
-import {Page} from '@playwright/test';
 import {AlertTestIdModifiers} from '@ironsource/fusion-ui/entities';
 import {getTestId, getTestIdSelector} from '../../global/utils';
-
-export class AlertComponent {
-    readonly page: Page;
-
-    constructor(page: Page) {
-        this.page = page;
+import {BaseComponent} from '../base-component';
+import {BaseElement} from '../../behavior';
+import {ButtonComponent} from '../button/button-component';
+
+export class AlertComponent extends BaseComponent {
+    private wrapperElement: BaseElement;
+    private messageElement: BaseElement;
+    private titleElement: BaseElement;
+    private actionButtonElement: ButtonComponent;
+    private closeButtonElement: ButtonComponent;
+
+    constructor(page, selector: string) {
+        super(page, selector);
+        this.wrapperElement = new BaseElement(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.WRAPPER)));
+        this.messageElement = new BaseElement(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.MESSAGE)));
+        this.titleElement = new BaseElement(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.TITLE)));
+        this.actionButtonElement = new ButtonComponent(
+            page,
+            getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.ACTION_BUTTON))
+        );
+        this.closeButtonElement = new ButtonComponent(page, getTestIdSelector(getTestId(this.selector, AlertTestIdModifiers.CLOSE_BUTTON)));
     }
 
     async waitForComponent({testId}: {testId: string}) {
         const loadedPageSelector = getTestIdSelector(getTestId(testId, AlertTestIdModifiers.WRAPPER));
-        await this.page.waitForSelector(loadedPageSelector);
+        await this.waitForSelector(loadedPageSelector);
     }
 
-    getAlertText({testId}: {testId: string}) {
-        return this.page.getByTestId(getTestId(testId, AlertTestIdModifiers.MESSAGE)).textContent();
+    async getAlertText() {
+        return this.messageElement.textContent();
```

Comment:
i get this error: Property 'textContent' does not exist on type 'ButtonComponent'.ts(2339)
I checked your buttonComponent class and it seems that the implementation there can use a bit modification.

- class Button should extend from base element (in order for the textContext method to be inherited)
-  class Button's properties should be button: Clickable, label: StaticText;

Reply:
I think already done, but lets discuss it

Reply:
Unify button and ButtonComponent class

Reply:
Done, also move all components from demand_platform_ui and fixed dependencies